python/cuml/neighbors/kneighbors_classifier.pyx

```diff
@@ -325,15 +318,12 @@ class KNeighborsClassifier(NearestNeighbors):
 
         self.handle.sync()
 
+        del knn_indices
```

Comment:
Same as above, I don't believe we need explicit deletes for locally scoped variables. They should disappear when the function returns.